File: src/main/java/me/austin/queer/modules/setting/settings/ToggleSetting.java
package me.austin.queer.modules.setting.settings;

import me.austin.queer.modules.setting.Setting;

public class ToggleSetting extends Setting<Boolean> {
    public ToggleSetting(String name, String description, boolean defaultValue) {
        super(name, description, defaultValue);
    }

    public void toggle() {
        this.set(!this.get());
    }
}

File: src/main/kotlin/trans/rights/client/modules/setting/settings/ToggleSetting.kt
package trans.rights.client.modules.setting.settings

import trans.rights.client.modules.setting.Setting

class ToggleSetting(name: String, description: String, value: Boolean, default: Boolean) : Setting<Boolean>(name, description, value, default) {
    fun toggle() {
        this.value = !this.value
    }
}